ts/features/itwallet/discovery/components/ItwPaywallComponent.tsx

```diff
@@ -0,0 +1,315 @@
+import {
+  Badge,
+  Body,
+  ContentWrapper,
+  Divider,
+  FeatureInfo,
+  H4,
+  H6,
+  IOColors,
+  IOIcons,
+  Icon,
+  VSpacer,
+  VStack,
+  useIOTheme
+} from "@pagopa/io-app-design-system";
+import { TxtLinkNode, TxtParagraphNode } from "@textlint/ast-node-types";
+import { useCallback, useState } from "react";
+import { FlatList, ListRenderItemInfo, StyleSheet, View } from "react-native";
+import {
+  Canvas,
+  LinearGradient,
+  RoundedRect,
+  vec
+} from "@shopify/react-native-skia";
+import ItwHero from "../../../../../img/features/itWallet/l3/itw_hero.svg";
+import IOMarkdown from "../../../../components/IOMarkdown";
+import {
+  getTxtNodeKey,
+  handleOpenLink
+} from "../../../../components/IOMarkdown/renderRules";
+import { Renderer } from "../../../../components/IOMarkdown/types";
+import I18n from "../../../../i18n";
+import { useIOSelector } from "../../../../store/hooks";
+import { tosConfigSelector } from "../../../tos/store/selectors";
+import { IOScrollView } from "../../../../components/ui/IOScrollView";
+import { useHeaderSecondLevel } from "../../../../hooks/useHeaderSecondLevel";
+import { emptyContextualHelp } from "../../../../utils/emptyContextualHelp";
+
+const markdownRules = {
+  Paragraph(paragraph: TxtParagraphNode, render: Renderer) {
+    return (
+      <Body color="white" key={getTxtNodeKey(paragraph)}>
+        {paragraph.children.map(render)}
+      </Body>
+    );
+  },
+  Link(link: TxtLinkNode, render: Renderer) {
+    return (
+      <Body
+        asLink
+        color="white"
+        weight="Semibold"
+        key={getTxtNodeKey(link)}
+        onPress={() => handleOpenLink(link.url)}
+      >
+        {link.children.map(render)}
+      </Body>
+    );
+  }
+};
+
+export type ItwPaywallComponentProps = {
+  onContinuePress: () => void;
+};
+
+export const ItwPaywallComponent = (_: ItwPaywallComponentProps) => {
+  const theme = useIOTheme();
+
+  const backgroundColor = IOColors[theme["appBackground-accent"]];
+
+  useHeaderSecondLevel({
+    backgroundColor,
+    contextualHelp: emptyContextualHelp,
+    supportRequest: true,
+    title: "",
+    variant: "contrast"
+  });
+
+  return (
+    <IOScrollView
+      includeContentMargins={false}
+      contentContainerStyle={{
+        flexGrow: 1
+      }}
+    >
+      <ContentWrapper
+        style={{
+          backgroundColor,
+          paddingVertical: "100%",
+          marginVertical: "-100%"
+        }}
+      >
+        <InnerComponent />
+      </ContentWrapper>
+    </IOScrollView>
+  );
+};
+
+const InnerComponent = () => {
+  const { tos_url } = useIOSelector(tosConfigSelector);
+
+  return (
+    <VStack space={24}>
+      <View style={styles.cardContainer}>
+        <BackgroundGradient />
+        <View style={{ paddingBottom: 24 }}>
+          <View style={{ alignItems: "center" }}>
+            <ItwHero width={286} height={192} />
```

Comment:
As per the Figma design, the hero should be edge to edge. We should consider removing the horizontal padding for the component. What do you think?